solution-base/mongodb/patches/mongodb-sharded-fix-podmonitor.patch

```diff
@@ -1,20 +1,21 @@
 diff --git a/solution-base/mongodb/charts/mongodb-sharded/templates/shard/shard-data-podmonitor.yaml b/solution-base/mongodb/charts/mongodb-sharded/templates/shard/shard-data-podmonitor.yaml
-index 3c689fc0..e43b5698 100644
+index d2c9c0cb..e6fa88c6 100644
 --- a/solution-base/mongodb/charts/mongodb-sharded/templates/shard/shard-data-podmonitor.yaml
 +++ b/solution-base/mongodb/charts/mongodb-sharded/templates/shard/shard-data-podmonitor.yaml
-@@ -1,6 +1,5 @@
+@@ -4,8 +4,7 @@ SPDX-License-Identifier: APACHE-2.0
+ */}}
+ 
  {{- if and .Values.shards .Values.metrics.enabled .Values.metrics.podMonitor.enabled }}
 -{{- $replicas := .Values.shards | int }}
 -{{- range $i, $e := until $replicas }}
 +{{- $i := 0 }}
  apiVersion: monitoring.coreos.com/v1
  kind: PodMonitor
  metadata:
-@@ -32,4 +31,3 @@ spec:
-     matchLabels: {{- include "common.labels.matchLabels" $ | nindent 6 }}
+@@ -37,6 +36,4 @@ spec:
+     matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" $podLabels "context" $ ) | nindent 6 }}
        app.kubernetes.io/component: shardsvr
- {{- end }}
+       shard: {{ $i | quote }}
```

Comment:
should not add a `shard` label filter, since we create a single podMonitor which matches every shard's pod : either we keep the loop (and need to filter on the shard), or we keep a single one and must not filter on the shard.

(later solution prefered, as it was before)

Reply:
this was not added manually , it's coming from the charts update ( using `make fetch-mongodb-sharded ` ) , the patch was just updated to apply the already existing changes , should we patch this as well ?

Reply:
yes, since we donnow have a single podMonitor, it **must not** filter on the shard...

alternatively, maybe the podMonitors are now properly generated, with one podMonitor per shard & proper filtering; and we could thus remove this patch altogether...

Reply:
patch updated here : https://github.com/scality/Zenko/pull/2028/commits/278f25041e230c660f3642820ad413f93a4e16fd#diff-bb1ea55e85cfff3a0fc51b35e678b12f7421adf03e1f4842710a3fe7f80f50a0R1-R24 
Indeed the updated chart creates one podmonitor/shard , went back to the previous solution : keeping a single one and not filtering on the shard